### ISSUE TITLE
Replication Push (to CouchDB) is failing when attachments are involved

### ISSUE THREAD

**Issue by borrrden** [couchbase/couchbase-lite-ios#817]
https://github.com/couchbase/couchbase-lite-net/issues/449

There was some stuff in here that was incorrect on .NET but not on iOS that can be ignored but the crux of the problem is that CouchDB potentially sends the revpos of an attachment in a "follows" style response, but Couchbase Lite only checks for revpos in a "stub" style response.  

Unit Test: https://github.com/couchbase/couchbase-lite-net/blob/master/src/Couchbase.Lite.Tests.Shared/ReplicationTest.cs#L409-L493

Solution is to add the following lines: https://github.com/couchbase/couchbase-lite-net/blob/master/src/Couchbase.Lite.Shared/Documents/AttachmentInternal.cs#L238-L245

**Comment by borrrden** [comment]
Oh, looks like I still have the Objective-C version of the test I used lying around:

https://gist.github.com/borrrden/c581d3ee45c75ffcfe8f

**Comment by hideki** [comment]
CBL java had issue of  push replication with attachment. This issue was fixed by community developer. This might help CBL iOS. https://github.com/couchbase/couchbase-lite-java-core/issues/600

**Comment by pasin** [comment]
I could reproduce the issue with @borrrden's unit test. I simplified the unit test by only creating one document. I ran the unit test on both CouchDb and Sync-Gateway. CouchDB returned an error as it couldn't find the (stub) attachment as the specified revpos. Sync-Gateway on the other hand didn't report the error.

```
SyncGateway:

Push #1:
{"_attachments":{"myattachment":{"content_type":"text/plain; charset=utf-8","digest":"sha1-oaDsDkUDon9gAShhMF9l7cxugW8=","follows":true,"length":51200,"revpos":1}},"_id":"-mNhwBuzUfcGaF44mcOcww2","_rev":"1-119725eeb7585545cc849fda550d09b8","_revisions":{"ids":["119725eeb7585545cc849fda550d09b8"],"start":1},"foo":"bar"}

Push #2:
{"docs":[{"foo":"bar","_id":"-mNhwBuzUfcGaF44mcOcww2","_rev":"2-a48e751aa24f7c501d91460fedc6a71e","_attachments":{"myattachment":{"stub":true,"revpos":1}},"_revisions":{"ids":["a48e751aa24f7c501d91460fedc6a71e","119725eeb7585545cc849fda550d09b8"],"start":2},"extraminutes":"5"}],"new_edits":false}

Pull:
{"_attachments":{"myattachment":{"content_type":"text/plain; charset=utf-8","digest":"sha1-oaDsDkUDon9gAShhMF9l7cxugW8=","follows":true,"length":51200,"revpos":1}},"_id":"-mNhwBuzUfcGaF44mcOcww2","_rev":"2-a48e751aa24f7c501d91460fedc6a71e","_revisions":{"ids":["a48e751aa24f7c501d91460fedc6a71e","119725eeb7585545cc849fda550d09b8"],"start":2},"extraminutes":"5","foo":"bar"}

Push #3:
{"docs":[{"foo":"bar","_id":"-mNhwBuzUfcGaF44mcOcww2","_rev":"3-f2df8fff594b9fa05aead40e3439a672","_attachments":{"myattachment":{"stub":true,"revpos":2}},"_revisions":{"ids":["f2df8fff594b9fa05aead40e3439a672","a48e751aa24f7c501d91460fedc6a71e"],"start":3},"extraminutes":"10"}],"new_edits":false}

CouchDB:

Push #1:
{"_attachments":{"myattachment":{"content_type":"text/plain; charset=utf-8","digest":"sha1-oaDsDkUDon9gAShhMF9l7cxugW8=","follows":true,"length":51200,"revpos":1}},"_id":"-5n2PDOwjOteSh5EtHnRi4U","_rev":"1-119725eeb7585545cc849fda550d09b8","_revisions":{"ids":["119725eeb7585545cc849fda550d09b8"],"start":1},"foo":"bar"}

Push #2:
{"docs":[{"foo":"bar","_id":"-5n2PDOwjOteSh5EtHnRi4U","_rev":"2-a48e751aa24f7c501d91460fedc6a71e","_attachments":{"myattachment":{"stub":true,"revpos":1}},"_revisions":{"ids":["a48e751aa24f7c501d91460fedc6a71e","119725eeb7585545cc849fda550d09b8"],"start":2},"extraminutes":"5"}],"new_edits":false}

Pull:
{"_id":"-5n2PDOwjOteSh5EtHnRi4U","_rev":"2-a48e751aa24f7c501d91460fedc6a71e","foo":"bar","extraminutes":"5","_revisions":{"start":2,"ids":["a48e751aa24f7c501d91460fedc6a71e","119725eeb7585545cc849fda550d09b8"]},"_attachments":{"myattachment":{"content_type":"text/plain; charset=utf-8","revpos":1,"digest":"md5-W3qvModIgI6HVCj3N0O4IQ==","length":51200,"follows":true,"encoding":"gzip","encoded_length":85}}}

Push#3:
{"docs":[{"foo":"bar","_id":"-5n2PDOwjOteSh5EtHnRi4U","_rev":"3-4741922c82e0464341898d939d02bad7","_attachments":{"myattachment":{"stub":true,"revpos":2}},"_revisions":{"ids":["4741922c82e0464341898d939d02bad7","a48e751aa24f7c501d91460fedc6a71e"],"start":3},"extraminutes":"10"}],"new_edits":false}

Error:
{"error":"missing_stub","reason":"id:-5n2PDOwjOteSh5EtHnRi4U, name:myattachment"}
```

**Comment by pasin** [comment]
@zgramana I have seen the same issue was in 1.1.1 for .NET so I will move the fix to 1.1.1 iOS as well.
